projects/ngrx.io/content/guide/eslint-plugin/rules/signal-store-feature-should-use-generic-type.md

```diff
@@ -10,3 +10,177 @@ A custom Signal Store feature that accepts an input should define a generic type
 
 <!-- Everything above this generated, do not edit -->
 <!-- MANUAL-DOC:START -->
+
+## Rule Details
+
+This rule ensure that Signal Store feature that accepts an input defines a generic type.
+
+Examples of **incorrect** code for this rule:
+
+<!-- prettier-ignore -->
+```ts
+// PROBLEM ❌
+const withY = () => signalStoreFeature(
+//                  ~~~~~~~~~~~~~~~~~~ [${messageId}]
+    { state: type<{ y: number }>() },
+    withState({})
+  );
+```
+
+```ts
+// PROBLEM ❌
+const withY = () => signalStoreFeature(
+//                  ~~~~~~~~~~~~~~~~~~ [${messageId}]
+  type<{ state: { y: number } }>(),
+  withState({})
+);
+```
+
+<!-- prettier-ignore -->
+```ts
+// PROBLEM ❌
+const withY = () => {
+  return signalStoreFeature(
+//       ~~~~~~~~~~~~~~~~~~ [${messageId}]
+    { state: type<{ y: number }>() },
+    withState({})
+  );
+}
+```
+
+<!-- prettier-ignore -->
+```ts
+// PROBLEM ❌
+const withY = () => {
+  return signalStoreFeature(
+//       ~~~~~~~~~~~~~~~~~~ [${messageId}]
+    type<{ state: { y: number } }>(),
+    withState({})
+  );
+}
+```
+
+<!-- prettier-ignore -->
+```ts
+// PROBLEM ❌
+const withY = () => {
+  return signalStoreFeature(
+//       ~~~~~~~~~~~~~~~~~~ [${messageId}]
+    type<{ state: { y: number } }>(),
+    withState({})
+  );
+}
+```
+
+<!-- prettier-ignore -->
+```ts
+// PROBLEM ❌
+function withY() {
+  return signalStoreFeature(
+//       ~~~~~~~~~~~~~~~~~~ [${messageId}]
+    type<{ state: { y: number } }>(),
+    withState({})
+  );
+}
+```
+
+<!-- prettier-ignore -->
+```ts
+// PROBLEM ❌
+function withY() {
+  return signalStoreFeature(
+//       ~~~~~~~~~~~~~~~~~~ [${messageId}]
+    { state: type<{ y: number }>() },
+    withState({})
+  );
+}
+```
+
+<!-- prettier-ignore -->
+```ts
+// PROBLEM ❌
+function withY() {
+  const feature = signalStoreFeature(
+//                ~~~~~~~~~~~~~~~~~~ [${messageId}]
+    { state: type<{ y: number }>() },
+    withState({})
+  );
+  return feature;
+}
+```
+
+Examples of **correct** code for this rule:
+
+<!-- prettier-ignore -->
+```ts
+// GOOD ✅
```

Comment:
I think we can remove the `Good` and `Problem` lines in the examples.
There's already a header that indicates the good and bad examples.

For the bad examples, it's important to indicate what the error is, which is what we do with `~~~~~` - this is the most important.

What do you think?
If you agree, we can also remove remove it for prefer-protected-state.

Reply:
Yes, this makes sense. I made kinda duplication. Will remove Good/Problem 👍